src/hooks/__tests__/use-load-data-pages.test.ts

```diff
@@ -0,0 +1,318 @@
+import * as nebula from "@nebula.js/stardust";
+import { renderHook } from "@testing-library/react";
+import React from "react";
+import { DEFAULT_PAGE_SIZE, Q_PATH } from "../../constants";
+import type { Model } from "../../types/QIX";
+import type { LayoutService, PageInfo, ViewService } from "../../types/types";
+import useLoadDataPages, { isMissingLayoutData, shouldFetchAdditionalData } from "../use-load-data-pages";
+
+const usePromiseResult: Promise<void> | undefined = Promise.resolve();
+let usePromiseError: Error | undefined;
+
+const usePromise = (callback: () => void) => {
+  callback();
+  return [usePromiseResult, usePromiseError];
+};
+
+jest.mock("@nebula.js/stardust", () => ({
```

Comment:
Should look into putting this under `<root-folder>/__mocks__/@nebula.js/stardust.ts` instead so that it is automatically mocked for all tests. Since we'll never be able to run nebula hooks anyway.